src/mol-model/structure/model/properties/utils/residue-set.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2021 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author David Sehnal <david.sehnal@gmail.com>
+ */
+
+import { StructureElement } from '../../../structure/element';
+import { StructureProperties } from '../../../structure/properties';
+
+export interface ResidueSetEntry {
+    label_asym_id: string,
+    label_comp_id: string,
+    label_seq_id: number,
+    label_alt_id: string,
+    ins_code: string,
+    // 1_555 by default
+    operator_name?: string
+}
+
+export class ResidueSet {
+    private index = new Map<string, Map<number, ResidueSetEntry[]>>();
+    private checkOperator: boolean = false;
+
+    add(entry: ResidueSetEntry) {
+        let root = this.index.get(entry.label_asym_id);
+        if (!root) {
+            root = new Map();
+            this.index.set(entry.label_asym_id, root);
+        }
+
+        let entries = root.get(entry.label_seq_id);
```

Comment:
Can this be extended to be with residues that don't have a label_seq_id (water, non-polymer, branched)?

Reply:
This should work for these to. They will have unique ``label_asym_id`` and ``label_seq_id`` will evaluate to 0. And the index contains ``ResidueSetEntry[]`` for each ``(asym_id, seq_id)`` pair, so I think it should work fine. Or am I missing something?

Reply:
I think it will break for branched entities, e.g., https://files.rcsb.org/view/1B5F.cif, multiple residues, single asym_id, no label_seq_id.

Reply:
![image](https://user-images.githubusercontent.com/21145776/111086470-8dd42080-851c-11eb-832a-5adad504c378.png)
![image](https://user-images.githubusercontent.com/21145776/111086489-99274c00-851c-11eb-9f18-c6bc8113e8da.png)

works fine.

Reply:
To compare, this is 5A residues surroundings without the struct_conn expand:

![image](https://user-images.githubusercontent.com/21145776/111086879-f3c1a780-851e-11eb-9430-ca2d47884665.png)

Reply:
Yeah, it works for this use case because you want the whole branched entity anyway and the residues in branched entities are connected with with struct con records as well.

I thing if you use ResiudueSet for another use case where you just want to add a single residue of a branched entity it will add all of them.